dev: Added SileroVadAlgorithm. Renamed VolumeAlgorithm to VolumeThresholdAlgorithm. Small improvements.

mishadobrits opened the pull request.

mishadobrits committed:
Added SileroVadAlgorithm.
Renamed VolumeAlgorithm to VolumeThresholdAlgorithm.
Small improvements.

gordeiegorov commented on Readme.md:
```diff
@@ -88,12 +104,16 @@ Testing process_one_video_in_computer function
 from main import process_one_video_in_computer
```
Добавить информацию по поводу того, что destination должно быть на диске C

gordeiegorov commented:
Сделать файл Example.py, в котором сделать самый простой (первый) пример - чтобы можно было сразу запустить и попробовать. И в readme об этом файлике написать

gordeiegorov reviewed (changes requested):
Доделать Readme и сделать Example.py

mishadobrits committed:
Renamed apply_calculated_interesting_and_boring_parts_to_video to apply_calculated_interesting_to_video.
Added complex algotithms: AlgNot, AlgAnd, AlgOr.
Add __str__ method to Algorithms.
In main functions 'print' command replaced using additional 'logger' parameter.

mishadobrits committed:
Upgraded existing algorithm. Now they are faster.

mishadobrits committed:
Small improvements.

mishadobrits committed:
Added wavio==0.0.4 to requirements.txt

mishadobrits committed:
Replace os.renames -> shutil.move in main functions.

mishadobrits committed:
Small crash fix and small Readme.md upgrade.

mishadobrits committed:
Upgrade WebRtcVADAlgorithm.__str__ method

mishadobrits committed:
Downscaled default "speed of skipped parts" for fixing rarely mkvmerge bug with "not increasing pts"

mishadobrits committed:
Still trying fix previous bug

mishadobrits committed:
Fixed desyncronization of audio and video, which occured after long time (about 30 minutes)

mishadobrits committed:
Fixed a small issue with desynchronization.
Fixed crash if input path contains a space.

mishadobrits committed:
Fixed a stupid bug in AlgNot.

mishadobrits committed:
Deleted one print.

mishadobrits committed:
Small change: create_valid_path is created as a separate function. It can be used by users for their needs.

mishadobrits committed:
An insignificant structure change of "create_valid_path" function.

mishadobrits committed:
Another insignificant structure change of "create_valid_path" function.

mishadobrits committed:
Another "create_valid_path" function preserves file extension.

mishadobrits committed:
Updated to new Silero API.

mishadobrits committed:
Upgraded numpy==1.20.2 -> numpy==1.21.4.

mishadobrits committed:
Added splitting audio in chunks and processing chunks (for decreasing memory) in SileroVad. This should be done in a first commit when added SileroVad, but whatever.

mishadobrits committed:
Added splitting audio in chunks and processing chunks (for decreasing memory) in SileroVad. This should be done in a first commit when added SileroVad, but whatever.

mishadobrits committed:
Using ffmpeg.exe from imageio-ffmpeg.

mishadobrits committed:
Removed the moviepy library from dependencies, used only its submodule imageio-ffmpeg.

mishadobrits committed:
The converting (.wav -> .aac) changed to facter (.wav -> .flac) before merging with video track.

mishadobrits committed:
The previous version extracted the audio from the media 2 times: for calculating interesting parts and for cutting the video.

mishadobrits committed:
Added support for audio samplerate greater than 48k.

mishadobrits committed:
Add subtitles algorithm. Decreased programm work time by doing some tasks in parallelling.

mishadobrits committed:
Removed one print

mishadobrits committed:
Fixed double execution of the "prepare_audio" function, if the original videopath contains spaces.

mishadobrits committed:
Added audio preprocessing as a parameter to the process_using_algorithm function.

mishadobrits committed:
Added streamlit example. Added librosa ro requirements.txt.

mishadobrits committed:
Added soundfile to requirements.txt.

mishadobrits committed:
A small change to make the streamlit work. Move 'import librosa' to only function where librosa used.

mishadobrits committed:
A small change to make the streamlit work. Added the mkvtoolnix downloading.

mishadobrits committed:
Add packages.txt. The streamlit (in cloud) might work after this commit.

mishadobrits committed:
Test for the streamlit cloud.

mishadobrits committed:
Test for the streamlit cloud.

mishadobrits committed:
Test for the streamlit cloud 6.

mishadobrits committed:
Test for the streamlit cloud 6.

mishadobrits committed:
Applying timecodes to video_track instead of applying them to 0-th track.

mishadobrits committed:
Created the new AlgAnd that (should be) a little faster and better. Previous AlgAnd renamed to AlgAnd1. You can specify the type of AlgAnd in the AlgOr.__init__ function (AlgOr is a Wrapper of AlgAnd and AlgNot)

mishadobrits committed:
Created the final version of the AlgAnd. Improvement in case if the input_video_path contains "'". Add few useless Algorithms.

mishadobrits committed:
Allow more versions of some packages in requirements.txt .

mishadobrits committed:
VFR video support. Improved Readme

mishadobrits committed:
Fixed bug 'Too many packets buffered for output stream'. Added LICENSE.

mishadobrits committed:
Nice output. Increased videochunk. Updated Readme.md.

mishadobrits committed:
Nice output. Increased videochunk. Updated Readme.md.

mishadobrits committed:
Nicer output.

mishadobrits committed:
Fixed bug with the lagging videotrack.
